### ISSUE TITLE
output.mem has a negative value

### ISSUE THREAD

**Issue by PythonGuython** [Herring-UGACSEE-4290/Project-3#7]
to reproduce:
assembly BabyTest.asm, view output.mem

here is the output:
```
@00000000
82 D0 00 00  
82 80 00 00  
00 00 00 00  
02 00 00 00  
23 8A 00 00  
33 8A 00 00  
35 40 00 01  
35 8A 00 00  
27 48 00 00  
37 48 00 00  
29 48 00 00  
2B 48 00 00  
7B 48 00 00  
08 00 00 00  
0A 00 00 00  
04 00 00 00  
06 00 00 00  
6C 00 00 00  
C0 00 00 20  
-0 00 00 4C  
C4 00 00 00  
C8 00 00 00  
82 D0 00 00  
82 80 00 00  
00 00 00 00  
02 00 00 00  
23 8A 00 00  
33 8A 00 00  
35 40 00 01  
35 8A 00 00  
27 48 00 00  
37 48 00 00  
29 48 00 00  
2B 48 00 00  
7B 48 00 00  
08 00 00 00  
0A 00 00 00  
04 00 00 00  
06 00 00 00  
6C 00 00 00  
C0 00 00 20  
-0 00 00 A4  
C4 00 00 00  
C8 00 00 00  
D0 00 00 00
```  
note the lines -0 00 00 A4  and -0 00 00 4C

I believe these are both branch instructions

**Comment by PythonGuython** [comment]
additional investigation:

```
start:
    B.eq start
```
^^ this does not recreate the issue

```
start:
    ADD r1,r2, #0x0
label:
    B.eq start
```
^^but this DOES recreate the issue

**Comment by MichaelStarks** [comment]
lable1:
    MOV r2, #0x3
    SUB r1, r3, #0x9
    ADD r0, r12, #0x3
    ADDS r0, r14, #0x5
;
thingy:
    CMP r1,r1, #0x40
    AND r4,r3, #0x32
    ANDS r4, r3, #0xa3
    or r4,r15, #0x4a
    ors r4,r15, #0x4a
    not r0, r3
    ;B lable1; check this for errors
    ;B #0x0000
    
    BR r4, #0xa5
    MOV32 r3, #0x0
    RMB
    XOR r1,r2,r3
    XORS r1,r2,r3
    ;FCB #0x04

test:
    B.eq thingy
    HALT

thingy
Label #: 16
Line #:  64

**Comment by MichaelStarks** [comment]
Fixed! 76c205ba3ff935744f56e219905d78d94648de7d

issue closed
